crates/primitives/src/bits/serde.rs

```diff
@@ -10,13 +14,83 @@ impl<const N: usize> Serialize for FixedBytes<N> {
 
 impl<'de, const N: usize> Deserialize<'de> for FixedBytes<N> {
     fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
-        hex::deserialize::<'de, D, [u8; N]>(deserializer).map(Self)
+        struct FixedVisitor<const N: usize>;
+
+        impl<'de, const N: usize> Visitor<'de> for FixedVisitor<N> {
+            type Value = FixedBytes<N>;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+                write!(
+                    formatter,
+                    "{} bytes, represented as an array of u8, or a hex string of length {}",
+                    N,
+                    N * 2
+                )
+            }
+
+            fn visit_bytes<E>(self, v: &[u8]) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                <[u8; N]>::try_from(v)
+                    .map(FixedBytes)
```

Comment:
maybe this should be done in hex? https://docs.rs/const-hex/latest/src/const_hex/serde.rs.html#60-97

Reply:
mmm. I didn't want to introduce non-hex behavior to the `const_hex` crate, which this does

This would be a good chance to reify our serialization strategy for these things. Something like:

Serializing
- if human readable serializer, always hex
- if non-human readable, always write bytes

Deserializing
- Accept bytearray/buf
- Accept Hex string with & w/o 0x prefix
- Accept sequence containing bytes

Does that seem reasonable?

Reply:
yeah makes sense